src/CVRF/cvrf_enumeration.c

```diff
@@ -0,0 +1,321 @@
+	{CVRF_PRODUCT_STATUS_RECOMMENDED, "Recommended"},
+	{CVRF_PRODUCT_STATUS_LAST_AFFECTED, "Last Affected"},
+	{0, NULL}
+};
+
+static cvrf_product_status_type_t cvrf_product_status_type_parse(xmlTextReaderPtr reader) {
+	return cvrf_enumeration_attr(reader, "Type", CVRF_PRODUCT_STATUS_TYPE_MAP);
+}
+
+const char *cvrf_product_status_type_get_text(cvrf_product_status_type_t product_status_type) {
+	return cvrf_enumeration_get_text(CVRF_PRODUCT_STATUS_TYPE_MAP, product_status_type);
+}
+
+
+const struct oscap_string_map CVRF_THREAT_TYPE_MAP[] = {
+	{CVRF_THREAT_IMPACT, "Impact"},
+	{CVRF_THREAT_EXPLOIT_STATUS, "Exploit Status"},
+	{CVRF_THREAT_TARGET_SET, "Target Set"},
+	{0, NULL}
+};
+
+static cvrf_threat_type_t cvrf_threat_type_parse(xmlTextReaderPtr reader) {
+	return cvrf_enumeration_attr(reader, "Type", CVRF_THREAT_TYPE_MAP);
+}
+const char *cvrf_threat_type_get_text(cvrf_threat_type_t threat_type) {
+	return cvrf_enumeration_get_text(CVRF_THREAT_TYPE_MAP, threat_type);
+}
+
+
+const struct oscap_string_map CVRF_REMEDIATION_TYPE_MAP[] = {
+	{CVRF_REMEDIATION_WORKAROUND, "Workaround"},
+	{CVRF_REMEDIATION_MITIGATION, "Mitigation"},
+	{CVRF_REMEDIATION_VENDOR_FIX, "Vendor Fix"},
+	{CVRF_REMEDIATION_NONE_AVAILABLE, "None Available"},
+	{CVRF_REMEDIATION_WILL_NOT_FIX, "Will Not Fix"},
+	{0, NULL}
+};
+
+static cvrf_remediation_type_t cvrf_remediation_type_parse(xmlTextReaderPtr reader) {
+	return cvrf_enumeration_attr(reader, "Type", CVRF_REMEDIATION_TYPE_MAP);
+}
+const char *cvrf_remediation_type_get_text(cvrf_remediation_type_t remediation_type) {
+	return cvrf_enumeration_get_text(CVRF_REMEDIATION_TYPE_MAP, remediation_type);
+}
+
+/*****************************************************************************
+ * CVRF Item Type
+ */
+
+struct cvrf_item_spec {
+	cvrf_item_type_t type;
+	const char *tag_name;
+	const char *container_name;
+};
+
+static const struct cvrf_item_spec CVRF_ITEM_TYPE_MAP[] = {
+	{CVRF_DOCUMENT_PUBLISHER, "DocumentPublisher", NULL},
+	{CVRF_DOCUMENT_TRACKING, "DocumentTracking", NULL},
+	{CVRF_REVISION, "Revision", "RevisionHistory"},
+	{CVRF_DOCUMENT_REFERENCE, "Reference", "DocumentReferences"},
+	{CVRF_ACKNOWLEDGMENT, "Acknowledgment", "Acknowledgments"},
+	{CVRF_PRODUCT_TREE, "ProductTree", NULL},
+	{CVRF_BRANCH, "Branch", NULL},
+	{CVRF_GROUP, "Group", "ProductGroups"},
+	{CVRF_RELATIONSHIP, "Relationship", NULL},
+	{CVRF_PRODUCT_NAME, "FullProductName", NULL},
+	{CVRF_VULNERABILITY, "Vulnerability", NULL},
+	{CVRF_INVOLVEMENT, "Involvement", "Involvements"},
+	{CVRF_SCORE_SET, "ScoreSet", "CVSSScoreSet"},
+	{CVRF_PRODUCT_STATUS, "Status", "ProductStatuses"},
+	{CVRF_THREAT, "Threat", "Threats"},
+	{CVRF_REMEDIATION, "Remediation", "Remediations"},
+	{CVRF_REFERENCE, "Reference", "References"},
+	{0, NULL, NULL}
+};
+
+static const struct cvrf_item_spec *cvrf_find_item_from_type(cvrf_item_type_t item_type) {
+	for (const struct cvrf_item_spec *mapptr = CVRF_ITEM_TYPE_MAP; mapptr->tag_name != NULL; ++mapptr) {
```

Comment:
Use oscap_string_to_enum and oscap_enum_to_string

This applies to other lines that convert enums to strings and back.